src/CBLTLSIdentity_Internal.hh

```
@@ -171,16 +177,141 @@ public:
+                                                          attributes,
+                                                          expiration);
+        assert(c4cert);
+
+#ifdef TARGET_OS_IPHONE
+        CBLError cblError;
+        if (!StripPublicKey(c4cert, &cblError)) {
+            C4Error::raise((C4ErrorDomain)cblError.domain, cblError.code, "Couldn't remove a public key");
+        }
+#endif
+        // Save the cert:
+        // Assertion: c4cert != nullptr
+        c4cert->save(false, persistentLabel);
+
+        CBL_Log(kCBLLogDomainListener, kCBLLogVerbose, "Created self signed cert(%.*s) isServer=%d expiry=%" PRId64 " attr=%s",
+                (int)persistentLabel.size, (char*)persistentLabel.buf, server, expiration, attributes.toJSONString().c_str());
+
+        return new CBLTLSIdentity(nullptr, new CBLCert(c4cert.get()));
+    }
+
+    static bool DeleteIdentityWithLabel(slice persistentLabel) {
+        std::scoped_lock<std::mutex> lock(_mutex);
+
+        // Load cert for getting the public key:
+        Retained<C4Cert> c4cert;
+        try {
+            c4cert = C4Cert::load(persistentLabel);
+        } catch (C4Error& err) {
+            //        if (!c4cert) {
+            if (err.code != 0 && err.code != kC4ErrorNotFound)
+                throw(err);
+            // Otherwise, no cert to delete.
+            return true;
+        }
+        if (!c4cert) return true;
+
+        // Get public key from the cert:
+        Retained<C4KeyPair> publicKey = c4cert->getPublicKey();
+        // No longer need c4cert
+
+        // Get private key from the public key:
+        Retained<C4KeyPair> persistentKey;
+        try {
+            persistentKey = C4KeyPair::persistentWithPublicKey(publicKey);
+        } catch (C4Error &err) {
+            if (err.domain != LiteCoreDomain || err.code !=  kC4ErrorNotFound) {
+                throw(err);
+            }
+            // Question: why ignore other errors?
```

I think it should return an error as well. Otherwise crash will happen?